spec/spec_helper.rb

```diff
@@ -3,23 +3,50 @@
   SimpleCov.start
 end
 
-Dir[Rails.root.join("spec/shared/**/*.rb")].each { |f| require f }
-Dir[File.join(__dir__, "support/**/*.rb")].each { |f| require f }
+Dir[Rails.root.join("spec/shared/**/*.rb")].sort.each { |f| require f }
+Dir[File.join(__dir__, "support/**/*.rb")].sort.each { |f| require f }
 
 require "manageiq-providers-ibm_cloud"
 
+# Iterate through the SSH keys and replace all values.
+# Being very careful with this one as fixing a data leak would be costly.
+def replace_ssh_keys(response)
+  data = JSON.parse(response.body, :symbolize_names => true)
+  return unless data.key?(:keys)
+
+  keys = {:fingerprint => 'SHA256:xxxxxxx', :public_key => 'RSA: VVVVVV'}
+  data.fetch(:keys).each_with_index do |v, i|
+    v.merge!(keys)
+    v[:name] = "random_key_#{i}"
+  end
+  response.body = data.to_json.force_encoding('ASCII-8BIT')
+end
+
+# Replace the contents of the token before writing to file.
+def replace_token_contents(response)
+  data = JSON.parse(response.body, :symbolize_names => true)
+  data.merge!({:refresh_token => 'REFRESH_TOKEN', :ims_user_id => '22222', :expiration => Date.new(2100, 1, 1).to_time.to_i})
+  response.body = data.to_json.force_encoding('ASCII-8BIT')
+end
+
+# Sanitize VPC VCR files.
+def vpc_sanitizier(interaction)
+  # Mask bearer token in recorded file.
+  interaction.request.headers['Authorization'] = 'Bearer xxxxxx' if interaction.request.headers.key?('Authorization')
+  # Replace IP V4 Addresses
+  interaction.response.body.gsub!(/([0-9]{1,3}\.){3}/, '127.0.0.')
+  # Replace ssh key data.
+  replace_ssh_keys(interaction.response)
+  interaction
+end
```

Comment:
Not for this PR, but we commonly move all this into a specific helper file such as spec/support/vcr_helper.rb

That being said we also have some common patterns for dealing with hiding credentials that we will probably want to follow for some consistency, though I'm not sure they can handle everything in this PR.

cc @agrare